### ISSUE TITLE
Initial setup fails when attempting to run in an ECS cluster

### ISSUE THREAD

**Issue by john-shaskin** [OctopusDeploy/Octopus-Docker#21]
We've been trying to get Octopus running as a task in ECS, and we're finding that it fails to complete the initial setup of the database and host configuration. We end up seeing the following sequence in the logs, as ECS attempts to spin up new tasks:

```
2018-05-09 22:14:50.3433   4384      1  INFO  ===================================================
2018-05-09 22:14:50.3433   4384      1  INFO  Testing connection to the SQL Server by attempting to connect to the 'master' database...
2018-05-09 22:14:52.7522   4384      1  INFO  Successfully connected to the SQL Server!
2018-05-09 22:14:53.3712   4384      1  INFO  Database 'octopusdeploy2' already exists on SQL Server at '****.us-west-2.rds.amazonaws.com', no need to create it.
2018-05-09 22:14:55.1502   4384      1  INFO  Checking the Octopus Master Key has been configured.
2018-05-09 22:14:55.1682   4384      1 ERROR  ===============================================================================
2018-05-09 22:14:55.2232   4384      1 FATAL  Key not valid for use in specified state.

System.Security.Cryptography.CryptographicException
   at System.Security.Cryptography.ProtectedData.Unprotect(Byte[] encryptedData, Byte[] optionalEntropy, DataProtectionScope scope)
   at Octopus.Shared.Configuration.FlatDictionaryKeyValueStore.Get[TData](String name, TData defaultValue, Nullable`1 protectionScope)
   at Octopus.Core.Initialization.BuiltIn.EnsureMasterKeyIsConfiguredInitializer.GenerateMasterKeyIfNotAlreadyConfigured(IMasterKeyConfiguration config, ILog log)
   at Nevermore.StoreInitializer.Initialize()
   at Octopus.Server.Commands.DatabaseCommand.Start()
   at Octopus.Shared.Startup.AbstractCommand.Start(String[] commandLineArguments, ICommandRuntime commandRuntime, OptionSet commonOptions)
   at Octopus.Shared.Startup.ConsoleHost.Run(Action`1 start, Action shutdown)
   at Octopus.Shared.Startup.OctopusProgram.Run()
2018-05-09 22:14:55.2472   4384      1 FATAL  -------------------------------------------------------------------------------
Terminating process with exit code 100
Full error details are available in the log files at:
C:\Octopus\Logs
C:\Users\ContainerAdministrator\AppData\Local\Octopus\Logs
If you need help, please send these log files to https://octopus.com/support
-------------------------------------------------------------------------------
```

Through my debugging, I've found that the `--tty` option seems to be key to getting the Octopus container to run and do its initial setup properly, when running it outside of ECS. Without the flag, the container stops partway through the setup, leaving things in a bad state. Running within ECS, I'm unaware of any way to enable the pseudo-TTY. So we're not able to get Octopus up and running in our ECS cluster.

**Comment by zentron** [comment]
@john-shaskin yes at the moment the `--tty` option is required since the process running inside the container is the exact same `Octopus.Server.exe` that  can be manually run from the command line (which itself waits for a readline to exit).
That being said in the exception you are getting seems like its at the point where its trying to verify the master-key. Are you setting the correct master-key in the environment variable list? 

Since this container image does not support multi-node HA is it possible to just run this on a standard, non-ECS cluster docker host?

**Comment by john-shaskin** [comment]
@zentron, thanks for the quick reply. With respect to the master-key, I've set up persistent volume mapping and a SQL server connection string. So on the initial running of the container, there is a master-key value getting generated and written into the c:\Octopus\OctopusServer.config file. Then, due to the lack of the `--tty` flag, the container stops, and ECS will try to spin up a new container with the same volume mapping and db connection string. And that's the point at which we get the error I posted above.

Once, I did try redeploying the ECS task, and supplying a master-key environment variable that matched the key that was persistently stored in the server config file, but I seemed to get the same result.

As far as not hosting this on an ECS cluster, it's definitely possible for us to do that. We've just been working on setting up a cluster to host various CI/CD tools, and were looking to evaluate whether the Octopus image would work in that environment. For now, we'll likely either take your suggestion on running a standalone Docker host, or just create an EC2 to run Octopus on.
